File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
  int minSubArrayLen(int target, vector<int>& nums) {
    int left = 0, ans = 100001;
    long long curr = 0;
    
    for (int right = 0; right < nums.size(); ++right) {
      curr += nums[right];
      
      while (curr >= target) {
        ans = min(ans, right - left + 1);
        curr -= nums[left++];
      }
    }
    
    return left > 0 ? ans : 0;
  }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
  public int minSubArrayLen(int target, int[] nums) {
    int left = 0, right = 0, curr = 0, ans = 100001;
    
    while (right < nums.length) {
      curr += nums[right];
      
      while (curr >= target) {
        ans = Math.min(ans, right - left + 1);
        curr -= nums[left];
        left++;
      }
      
      right++;
    }
    
    return ans != 100001 ? ans: 0;
  }
}